### PR TITLE
Build Xtensa kernels for Vision P6

### PR DESCRIPTION
**Pull request by advaitjain**
This PR currently includes commits from #45619.

Confirmed that all tests pass for vision P6:
 ```
make -f tensorflow/lite/micro/tools/make/Makefile TARGET=xtensa OPTIMIZED_KERNEL_DIR=xtensa TARGET_ARCH=vision_p6 XTENSA_CORE=P6_200528 test
```

Once this PR is merged we can turn on continuous builds for P6.

### CODE REVIEW

**Comment by advaitjain** [comment]
tagging @pnikam-cad @nyadla-sys @kpraving

**Review by njeffrie — COMMENTED** [review]
I'm going to wait until #45618 and this is rebased. Seems like that will provide a clearer picture of the delta here.

**Comment by advaitjain** [comment]
> I'm going to wait until #45618 and this is rebased. Seems like that will provide a clearer picture of the delta here.

The quantize refactor PR is merged so this PR is ready for review now.

File: tensorflow/lite/micro/kernels/BUILD [hunk 1]
```diff
@@ -183,6 +187,24 @@ cc_library(
...
+        ],
+        ":xtensa_hifimini": [
+            "xtensa/xtensa.h",
+        ],
+    }),
+    copts = micro_copts(),
+    deps = select({
+        "//conditions:default": [],
+        ":xtensa_hifimini": [
+            #"//third_party/xtensa/cstub64s:hifi_mini",
```
Comment (njeffrie) [inline tensorflow/lite/micro/kernels/BUILD:203]: What is the purpose of this commented line?

Comment (advaitjain) [reply tensorflow/lite/micro/kernels/BUILD:203]: This is automatically uncommented by copybara. Allows us the use of the proprietary cstubs internally while not breaking the build externally.

**Review by njeffrie — APPROVED** [review]

**Comment by advaitjain** [comment]
copybara import error occurred. retrying with "kokoro:force-run" and "ready to pull" labels.

**Pull request merged**